tests/e2e/basic/_basic_test_setup/setup__e2e_resources.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+
+@pytest.mark.sanity
+@pytest.mark.bridge
+@pytest.mark.nat
+@pytest.mark.vlan
+@pytest.mark.firmware
+class TestFirmware(object):
+
+    @pytest.mark.firmware_create
+    def test_firmware_create(self, upload_firmware, instantiate_testrail, instantiate_project, test_cases):
```

Comment:
I would rename instantiate_testrail to something more generic

Reply:
what are your suggestions?
i am thinking of reporting

Reply:
Reporting makes sense